### PR TITLE
Prepare FloatingPanel 2

### PR DESCRIPTION
**Pull request by SCENEE**
## Feature updates

* Support top, left and right positions of the panel
* Add ObjC compatibility
* Improve the layout customization
* Improve the behavior customization to handle it like UIScrollView
* Improve the remove interaction
* Fix many issues depending on API design

## API updates
* FloatingPanelController
    * Replace `surfaceOriginY` with `surfaceLocation`.
    * Enable to update `layout` and `behavior` directly.
    * Update scroll tracking API a bit to support multiple scroll view tracking in the future.
* FloatingPanelControllerDelegate
    * Change `floatingPanelDidMove` behavior which is also called in the spring animation.
    * Update removal interaction delegate 
        * Enable a panel to invoke the removal action at any state(position)
        * Add ` floatingPanel(_:shouldRemoveAt:with:)` to determine whether it invokes the removal interaction.
   * Add `floatingPanelDidEndDragging(_ vc:willAttract:)`
* Layout
    * Introduce LayoutAnchor for the more flexible layout.
        * Add a factional panel position in the FloatingPanelController view
        * Replace `FloatingPanelFullScreenLayout` and `FloatingPanelIntrinsicLayout`
* Behavior
    * Introduce the new spring effect using Numeric springing
    * Add properties to contorl numeric springing 
    * Remove .topInteractionBuffer/.bottomInteractionBuffer
        * You can control the panel max/min position in floatingPanel(_:didMove:) delegate method
* SurfaceView
    * Use`SurfaceAppearnace` to specify the rounding corners, shadow and background color.
* BackdropView
    * Disable the dismissal action of backdrop by default
        * You can enable it to set `fpc.backdropView.dismissalTapGestureRecognizer.isEnabled = true`
* FloatingPanelPanGestureRecognizer
    * Add `delegateProxy` to intercept the gesture recognizer delegate.
* Improve API naming
    * FloatingPanelPosition -> FloatingPanelState: Now FloatingPanelPosition is used to specify the panel positioning in a screen.
    * FloatingPanelGrabberHandleView -> GrabberView: Swift only
    * FloatingPanelSurfaceView  -> SurfaceView: Swift only
    * FloatingPanelBackdropView -> BackdropView: Swift only
    * Decelerating -> Attracting: Because the spring behavior is actually magnetic.
* Misc
    * Add UISpringTimingParameters(decelerationRate:frequencyResponse:initialVelocity:)

## Environment updates
* Swift 5.0 or later
* iOS 11 or later(iOS 10 might be working)
* Update the directory structure and file names

## TODO
- [x] Write Migration Guide to v2
- [x] Check README
- [x] Enhance the doc comments

### CODE REVIEW

**Comment by warpling** [comment]
Incredible. Can't wait for this!

**Comment by zntfdr** [comment]
This is awesome work! Thank you @SCENEE 🙏

File: Documentation/FloatingPanel 2.0 Migration Guide.md [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+
+### New `FloatingPanelLayoutAnchoring` classes
+
+The following objects adopting `FloatingPanelLayoutAnchoring` protocol are added to configure the flexible and explicit layout.
+
+#### `FloatingPanelLayoutAnchor`
+
+This class is used to specify a panel layout using insets from a rectangle area of the superview or safe area.
+
+* `FloatingPanelFullScreenLayout` is replaced with anchors uing `.superview` reference guide.
```
Comment (zntfdr) [inline Documentation/FloatingPanel 2.0 Migration Guide.md:134]:
```suggestion
* `FloatingPanelFullScreenLayout` is replaced with anchors using `.superview` reference guide.
```

File: Documentation/FloatingPanel 2.0 Migration Guide.md [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+* `interactionAnimator(_:to:with:)`, `moveAnimator(_:from:to:)`
+  * They are removed because the animators are replaced with the new spring effect.
+* `removalVelocity`, `removalProgress`
+  * They are replaced with `floatingPanel(_:shouldRemoveAt:with:)` of `FloatingPanelControllerDelegate`
+* `removalInteractionAnimator(_:with:)`
+  * It is integrated with `floatingPanel(_:animatorForDismissingWith:)` of `FloatingPanelControllerDelegate`.
+
+## `SurfaceView`
+
+* `SurfaceAppearnace` class and `SurfaceView.appearance` property are added to specify the rounding corners, shadows and background color.
```
Comment (zntfdr) [inline Documentation/FloatingPanel 2.0 Migration Guide.md:238]:
```suggestion
* `SurfaceAppearance` class and `SurfaceView.appearance` property are added to specify the rounding corners, shadows and background color.
```

File: Documentation/FloatingPanel 2.0 Migration Guide.md [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+    fpc.behavior = SearchPaneliPadBehavior()
+    fpc.panGestureRecognizer.delegateProxy = self
+}
+
+func gestureRecognizer(_ gestureRecognizer: UIGestureRecognizer, shouldRequireFailureOf otherGestureRecognizer: UIGestureRecognizer) -> Bool {
+    return false
+}
+```
+
+## Miscellanea
```
Comment (zntfdr) [inline Documentation/FloatingPanel 2.0 Migration Guide.md:291]:
```suggestion
## Miscellaneous
```

Comment (zntfdr) [reply Documentation/FloatingPanel 2.0 Migration Guide.md:291]: Miscellanea is correct, however I believe `Miscellaneous` fits better for this documentation 👍

**Comment by SCENEE** [comment]
Thank you so much, @zntfdr! I revised the migration guide following your reviews.

**Comment by zntfdr** [comment]
You're very welcome @SCENEE, can't wait to try this out 😃

**Comment by SCENEE** [comment]
Today I released version 2.0.0. Thank you so much, @zntfdr and everyone.

**Comment by aaronbrethorst** [comment]
Bravo @SCENEE, @zntfdr, and everyone else! 👏

**Pull request merged**